src/powershell-native/Install-PowerShellRemoting.ps1

```diff
@@ -103,84 +103,103 @@ function Generate-PluginConfigFile
     Write-Verbose "Created Plugin Config File: $pluginFile" -Verbose
 }
 
-######################
-#                    #
-# Install the plugin #
-#                    #
-######################
+function Install-PluginEndpoint {
+    param (
+        [switch]
+        # [parameter()]
+        $VersionIndependent
+    )
 
-if (! ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator"))
-{
-    Write-Error "WinRM registration requires Administrator rights. To run this cmdlet, start PowerShell with the `"Run as administrator`" option."
-    Break
-}
+    ######################
+    #                    #
+    # Install the plugin #
+    #                    #
+    ######################
 
-if ($PsCmdlet.ParameterSetName -eq "ByPath")
-{
-    $targetPsHome = $PowerShellHome
-    $targetPsVersion = & "$targetPsHome\pwsh" -NoProfile -Command '$PSVersionTable.PSVersion.ToString()'
-}
-else
-{
-    ## Get the PSHome and PSVersion using the current powershell instance
-    $targetPsHome = $PSHOME
-    $targetPsVersion = $PSVersionTable.PSVersion.ToString()
-}
+    if (! ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator"))
+    {
+        Write-Error "WinRM registration requires Administrator rights. To run this cmdlet, start PowerShell with the `"Run as administrator`" option."
+        Break
+    }
+
+    if ($PsCmdlet.ParameterSetName -eq "ByPath")
+    {
+        Write-Host "In By Path"
+        $targetPsHome = $PowerShellHome
+        $targetPsVersion = & "$targetPsHome\pwsh" -NoProfile -Command '$PSVersionTable.PSVersion.ToString()'
+    }
+    else
+    {
+        ## Get the PSHome and PSVersion using the current powershell instance
+        $targetPsHome = $PSHOME
+        $targetPsVersion = $PSVersionTable.PSVersion.ToString()
+    }
+
+    if ($VersionIndependent) {
```

Comment:
Please add a comment that illustrates the effect of this, i.e., before and after strings.

Reply:
Comment added.